tensorflow-framework/src/main/java/org/tensorflow/framework/optimizers/Optimizer.java

```diff
@@ -280,12 +271,68 @@ protected Op finish(List<Op> updateOperations, String name) {
   }
 
   /**
-   * Get the Name of the optimizer.
+   * Gets the Name of the optimizer.
    *
    * @return The optimizer name.
    */
   public abstract String getOptimizerName();
 
+  /**
+   * Sets the learning rate
+   *
+   * @param learningRate the learning rate
+   */
+  public final void setLearningRate(float learningRate) {
+    if (this.learningRatePlaceholder == null) {
```

Comment:
Everything seems to have grown a `this` reference. I don't think that's particularly necessary in these methods, as the argument could be `newLearningRate` rather than `learningRate` and then there is no aliasing.

Reply:
I do this out of habit. I can easily change it as you suggest.

Reply:
changed `setLearningRate` to `setLearningRate(float newLearningRate)`, removed spurious` this.`.

Update pushed